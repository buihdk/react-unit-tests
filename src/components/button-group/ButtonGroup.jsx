import React, { memo } from 'react';
import PropTypes from 'prop-types';
import I18n from 'i18n-js';
import { Button } from 'antd';
import classnames from 'classnames';

import { noop } from 'src/utils';

const styles = {};

export const ButtonGroup = ({
  cancelText,
  handleCancel,
  handleSubmit,
  submitText,
  isSubmitForm,
  customGroupBtnWrap,
}) => {
  let submitButtonCustom = {
    type: 'primary',
    className: styles.btnSize,
  };

  // TODO: check if submit type is Form then handle submit form
  // Otherwise, handle event submit
  if (isSubmitForm) {
    submitButtonCustom = { ...submitButtonCustom, htmlType: 'submit' };
  }
  submitButtonCustom = { ...submitButtonCustom, onClick: handleSubmit };

  return (
    <div className={classnames(styles.groupBtnWrap, customGroupBtnWrap)}>
      <Button
        onClick={handleCancel}
        className={classnames(styles.cancelBtn, styles.btnSize)}
      >
        {I18n.t(cancelText)}
      </Button>
      <Button {...submitButtonCustom}>{I18n.t(submitText)}</Button>
    </div>
  );
};

ButtonGroup.propTypes = {
  handleCancel: PropTypes.func,
  cancelText: PropTypes.string,
  submitText: PropTypes.string,
  handleSubmit: PropTypes.func,
  isSubmitForm: PropTypes.bool,
  customGroupBtnWrap: PropTypes.string,
};

ButtonGroup.defaultProps = {
  cancelText: '',
  submitText: '',
  handleCancel: noop,
  handleSubmit: noop,
  isSubmitForm: false,
  customGroupBtnWrap: '',
};

export default memo(ButtonGroup);
