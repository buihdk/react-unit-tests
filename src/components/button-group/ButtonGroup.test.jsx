import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { ButtonGroup } from './ButtonGroup';

describe(`ButtonGroup`, () => {
  const props = {
    cancelText: 'common.cancel',
    handleCancel: () => {},
    handleSubmit: () => {},
    submitText: 'common.save',
  };
  test(`renders without crashing`, () => {
    const wrapper = shallow(<ButtonGroup {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test(`renders without crashing with submit form`, () => {
    const newProps = {
      ...props,
      isSubmitForm: true,
    };
    const wrapper = shallow(<ButtonGroup {...newProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
