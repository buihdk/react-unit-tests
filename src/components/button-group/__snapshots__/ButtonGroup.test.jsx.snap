// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders without crashing 1`] = `
<div
  className=""
>
  <Button
    block={false}
    className=""
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
  >
    [missing "en.common.cancel" translation]
  </Button>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    type="primary"
  >
    [missing "en.common.save" translation]
  </Button>
</div>
`;

exports[`ButtonGroup renders without crashing with submit form 1`] = `
<div
  className=""
>
  <Button
    block={false}
    className=""
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
  >
    [missing "en.common.cancel" translation]
  </Button>
  <Button
    block={false}
    ghost={false}
    htmlType="submit"
    loading={false}
    onClick={[Function]}
    type="primary"
  >
    [missing "en.common.save" translation]
  </Button>
</div>
`;
